sleep/compile_error.go

```diff
@@ -0,0 +1,6 @@
+// +build go1.17
+
+package sleep
+
+// Force a compile error to "output" a message.
+const compileError int = "This package requires manual checks before it can be built on Go 1.17+. See https://github.com/getlantern/netstack/pull/6"
```

Comment:
I'm curious what you all think of this.  The goal is that somebody building a dependent package (e.g. flashlight) would see this:

```
# github.com/google/netstack/tcpip/link/rawfile
tcpip/link/rawfile/compile_error.go:6:7: cannot use "This package requires manual checks before it can be built on Go 1.17+. See https://github.com/getlantern/netstack/pull/6" (type untyped string) as type int
tcpip/link/rawfile/rawfile_unsafe.go:137:10: undefined: BlockingPoll
tcpip/link/rawfile/rawfile_unsafe.go:159:10: undefined: BlockingPoll
tcpip/link/rawfile/rawfile_unsafe.go:189:14: undefined: BlockingPoll
# github.com/google/netstack/tcpip/stack
tcpip/stack/linkaddrcache.go:92:14: undefined: sleep.Waker
tcpip/stack/linkaddrcache.go:125:40: undefined: sleep.Waker
tcpip/stack/linkaddrcache.go:183:132: undefined: sleep.Waker
tcpip/stack/linkaddrcache.go:234:65: undefined: sleep.Waker
tcpip/stack/registration.go:460:108: undefined: sleep.Waker
tcpip/stack/registration.go:463:60: undefined: sleep.Waker
tcpip/stack/route.go:117:32: undefined: sleep.Waker
tcpip/stack/route.go:142:36: undefined: sleep.Waker
tcpip/stack/stack.go:1131:127: undefined: sleep.Waker
tcpip/stack/stack.go:1146:75: undefined: sleep.Waker
tcpip/stack/stack.go:1146:75: too many errors
```

Instead of just the "undefined" errors without context.

Reply:
Looks good. I can't find a way to do this with some Go compiler directive. I'm sure something existed, but it does seem like it would be out of line with Go conventions.

Reply:
Yeah I wasn't able to find any such directive and I agree it would be out of line with Go conventions.